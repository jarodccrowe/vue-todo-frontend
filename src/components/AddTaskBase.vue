<template>
  <div>
    <font-awesome-icon @click="openForm" v-show="!formOpen" class="pointer" size="3x" :icon="['far', 'plus-square']"/>
    <b-form @submit="onSubmit" v-show="formOpen">
      <b-form-input
          id="input-1"
          v-model="form.name"
          type="text"
          required
          placeholder="What do you need to do?"
      ></b-form-input>
      <b-form-group>
        <b-form-radio v-model="form.type" name="some-radios" value="task">Task</b-form-radio>
        <b-form-radio v-model="form.type" name="some-radios" value="event">Event</b-form-radio>
      </b-form-group>
      <b-button type="submit" variant="primary">Submit</b-button>
      <b-button type="button" variant="secondary" @click="closeForm">Close</b-button>
    </b-form>
  </div>
</template>

<script>
import { mapState } from 'vuex'
export default {
  name: 'AddTaskBase',
  data() {
    return {
      form: {
        name: '',
        type: '',
      }
    }
  },
  computed: {
    ...mapState([
      'formOpen',
    ]),
  },
  methods: {
    openForm() {
      this.$store.dispatch('updateAddTaskFormOpen', true)
    },
    closeForm() {
      this.$store.dispatch('updateAddTaskFormOpen', false)
    },
    onSubmit(evt) {
      evt.preventDefault()
      this.$store.dispatch('addTask', this.form)
    },
  }
}
</script>

<style scoped>
</style>
