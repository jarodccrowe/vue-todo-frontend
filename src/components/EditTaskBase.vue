<template>
  <div>
    <b-form @submit="onSubmit" class="row no-gutters">
      <div class="col-11 offset-1 card p-2">
        <b-form-input
          id="editInput"
          class="form-control-sm"
          v-model="form.name"
          type="text"
          required
          placeholder="What is your new task?"
          v-focus
        ></b-form-input>
        <div class="row mt-2">
          <div class="text-left col-6">
            <b-button size="sm" type="submit" variant="primary">Update Task</b-button>
          </div>
          <div class="text-right col-6">
            <font-awesome-icon
              @click="closeForm"
              class="pointer mt-1"
              :icon="['far', 'times-circle']"
              size="lg"
            />
          </div>
        </div>
      </div>
    </b-form>
  </div>
</template>

<script>
import { mapState } from 'vuex'
const focus = {
  inserted(el) {
    el.focus()
  },
}
export default {
  name: 'EditTaskBase',
  props: {
    id: Number,
  },
  data() {
    return {
      form: {
        name: '',
      }
    }
  },
  directives: { focus },
  computed: {
    ...mapState([
      'formOpen',
    ]),
  },
  methods: {
    closeForm() {
      this.$store.dispatch('updateEditTaskFormOpen', { id: this.id, isOpen: false } )
    },
    onSubmit(evt) {
      evt.preventDefault()
      this.$store.dispatch('updateTask', { ...this.form, id: this.id})
      this.closeForm()
    },
  },
}
</script>

<style scoped>
</style>
