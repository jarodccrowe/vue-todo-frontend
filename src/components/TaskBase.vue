<template>
  <div>
    {{type}}
    {{name}}
    {{complete ? 'yes' : 'no'}}
    <button @click="updateTask({ id, complete: !complete })">Complete</button>
    <button @click="deleteTask(id)">Delete</button>
  </div>
</template>

<script>
import { mapActions } from 'vuex'
export default {
  name: 'TaskBase',
  props: {
    name: String,
    type: String,
    complete: Boolean,
    id: Number,
  },
  methods: mapActions([
    'updateTask',
    'deleteTask',
  ])
}
</script>

<style scoped>
</style>
