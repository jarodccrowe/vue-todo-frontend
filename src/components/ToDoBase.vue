<template>
  <div>
    <h5>
      ToDo List
    </h5>
    <div>
      {{ new Date() | moment("dddd, MMMM Do YYYY") }}
    </div>
    <hr />
    <div>
      <div class="text-center pt-3" v-if="loading">
        <b-spinner small label="Small Spinner" type="grow" class="text-info"></b-spinner>
      </div>
      <div>
        <div v-show="incompleteTasks.length === 0" class="text-muted">
          <small>
            <i>All tasks complete</i>
          </small>
        </div>
        <div v-for="task in incompleteTasks" v-bind:key="task.id" class="mb-2">
          <TaskBase v-bind="task" />
        </div>
      </div>
      <hr />
      <div class="incomplete-task-list">
        <div v-for="task in completeTasks" v-bind:key="task.id" class="mb-2">
          <TaskBase v-bind="task" />
        </div>
      </div>
      <hr />
      <div>
        <AddTaskBase />
      </div>
    </div>
  </div>
</template>

<script>
import { mapState } from 'vuex'
import TaskBase from './TaskBase.vue'
import AddTaskBase from './AddTaskBase.vue'

export default {
  name: 'ToDoBase',
  props: {
    deleteMe: String
  },
  components: {
    TaskBase,
    AddTaskBase,
  },
  computed: {
    ...mapState([
      'loading',
    ]),
    incompleteTasks () {
      return this.$store.getters.incompleteTasks
    },
    completeTasks () {
      return this.$store.getters.completeTasks
    },
  }
}
</script>

<style scoped>
h3 {
  margin: 40px 0 0;
}
ul {
  list-style-type: none;
  padding: 0;
}
li {
  display: inline-block;
  margin: 0 10px;
}
a {
  color: #04b2e0;
}
.incomplete-task-list {
  opacity: 0.5;
}
</style>
