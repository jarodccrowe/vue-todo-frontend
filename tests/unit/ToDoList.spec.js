import { mount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex';

import BootstrapVue from 'bootstrap-vue/dist/bootstrap-vue.esm';
import Moment from 'vue-moment';
import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome'
import { library } from '@fortawesome/fontawesome-svg-core'
import { faPen } from '@fortawesome/free-solid-svg-icons'
import {
  faSquare,
  faCheckSquare,
  faTrashAlt,
  faPlusSquare,
  faTimesCircle,
  faCheckCircle,
} from '@fortawesome/free-regular-svg-icons'
import 'bootstrap/dist/css/bootstrap.css';
import 'bootstrap-vue/dist/bootstrap-vue.css';

library.add(
  faSquare,
  faCheckSquare,
  faTrashAlt,
  faPen,
  faPlusSquare,
  faTimesCircle,
  faCheckCircle,
)

import ToDoBase from '@/components/ToDoBase.vue'
import AddTaskBase from '@/components/AddTaskBase.vue'
import EditTaskBase from '@/components/EditTaskBase.vue'
import TaskBase from '@/components/TaskBase.vue'

const localVue = createLocalVue()

localVue.use(Vuex)
localVue.use(Moment);
localVue.use(BootstrapVue);

localVue.component('font-awesome-icon', FontAwesomeIcon)

localVue.filter('capitalize', function (value) {
  if (!value) return ''
  value = value.toString()
  return value.charAt(0).toUpperCase() + value.slice(1)
})

let store

store = new Vuex.Store({
  state: {
    tasks: [],
    openEditTaskForms: [],
    loading: false,
    formOpen: true,
  },
  getters: {
    completeTasks: () => {
      return [
        {
          "id": 1,
          "name": "Walk the dog",
          "complete": true,
          "type": "task"
        },
      ]
    },
    incompleteTasks: () => {
      return [
        {
          "id": 2,
          "name": "Walk the dog again",
          "complete": false,
          "type": "task"
        },
      ]
    }
  }
})

describe('AddTaskBase.vue', () => {
  const wrapper = mount(AddTaskBase, {
    localVue,
    store,
  })
  it('Matches snapshot', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})

describe('EditTaskBase.vue', () => {
  const wrapper = mount(EditTaskBase, {
    localVue,
    store,
  })
  it('Matches snapshot', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})

describe('TaskBase.vue', () => {
  const wrapper = mount(TaskBase, {
    localVue,
    store,
  })
  it('Matches snapshot', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})

describe('ToDoBase.vue', () => {
  const wrapper = mount(ToDoBase, {
    localVue,
    store,
  })
  it('Matches snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  })
  it('Loads a task', () => {
    expect(wrapper.html()).toContain('Walk the dog again')
  })
})
