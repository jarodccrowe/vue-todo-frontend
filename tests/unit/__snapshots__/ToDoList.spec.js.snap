// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTaskBase.vue Matches snapshot 1`] = `
<div>
  <div
    class="text-md-left text-center"
  >
    <svg
      aria-hidden="true"
      class="pointer add-button svg-inline--fa fa-plus-square fa-w-14 fa-2x"
      data-icon="plus-square"
      data-prefix="far"
      focusable="false"
      role="img"
      style="display: none;"
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        class=""
        d="M352 240v32c0 6.6-5.4 12-12 12h-88v88c0 6.6-5.4 12-12 12h-32c-6.6 0-12-5.4-12-12v-88h-88c-6.6 0-12-5.4-12-12v-32c0-6.6 5.4-12 12-12h88v-88c0-6.6 5.4-12 12-12h32c6.6 0 12 5.4 12 12v88h88c6.6 0 12 5.4 12 12zm96-160v352c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V80c0-26.5 21.5-48 48-48h352c26.5 0 48 21.5 48 48zm-48 346V86c0-3.3-2.7-6-6-6H54c-3.3 0-6 2.7-6 6v340c0 3.3 2.7 6 6 6h340c3.3 0 6-2.7 6-6z"
        fill="currentColor"
      />
    </svg>
  </div>
   
  <form
    class="mt-2"
  >
    <input
      aria-required="true"
      class="mb-2 form-control"
      id="__BVID__6"
      placeholder="What do you need to do?"
      required="required"
      test-data-name=""
      type="text"
    />
     
    <div
      class="mb-2 custom-control custom-radio"
    >
      <input
        aria-required="true"
        autocomplete="off"
        class="custom-control-input"
        id="__BVID__7"
        name="some-radios"
        required="required"
        test-data-radiotask=""
        type="radio"
        value="task"
      />
      <label
        class="custom-control-label"
        for="__BVID__7"
      >
        Task
      </label>
    </div>
     
    <div
      class="mb-2 custom-control custom-radio"
    >
      <input
        aria-required="true"
        autocomplete="off"
        class="custom-control-input"
        id="__BVID__8"
        name="some-radios"
        required="required"
        type="radio"
        value="event"
      />
      <label
        class="custom-control-label"
        for="__BVID__8"
      >
        Event
      </label>
    </div>
     
    <div
      class="row mt-3 no-gutters"
    >
      <div
        class="col"
      >
        <button
          class="btn btn-sm btn-primary"
          type="submit"
        >
          Add Task
        </button>
      </div>
       
      <div
        class="text-right col"
      >
        <svg
          aria-hidden="true"
          class="pointer svg-inline--fa fa-times-circle fa-w-16 fa-2x"
          data-icon="times-circle"
          data-prefix="far"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            class=""
            d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200zm101.8-262.2L295.6 256l62.2 62.2c4.7 4.7 4.7 12.3 0 17l-22.6 22.6c-4.7 4.7-12.3 4.7-17 0L256 295.6l-62.2 62.2c-4.7 4.7-12.3 4.7-17 0l-22.6-22.6c-4.7-4.7-4.7-12.3 0-17l62.2-62.2-62.2-62.2c-4.7-4.7-4.7-12.3 0-17l22.6-22.6c4.7-4.7 12.3-4.7 17 0l62.2 62.2 62.2-62.2c4.7-4.7 12.3-4.7 17 0l22.6 22.6c4.7 4.7 4.7 12.3 0 17z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
  </form>
</div>
`;

exports[`EditTaskBase.vue Matches snapshot 1`] = `
<div>
  <form
    class="row no-gutters"
  >
    <div
      class="col-11 offset-1 card p-2"
    >
      <input
        aria-required="true"
        class="form-control-sm form-control"
        id="editInput"
        placeholder="What is your new task?"
        required="required"
        type="text"
      />
       
      <div
        class="row mt-2"
      >
        <div
          class="text-left col-6"
        >
          <button
            class="btn btn-primary btn-sm"
            type="submit"
          >
            Update Task
          </button>
        </div>
         
        <div
          class="text-right col-6"
        >
          <svg
            aria-hidden="true"
            class="pointer mt-1 svg-inline--fa fa-times-circle fa-w-16 fa-lg"
            data-icon="times-circle"
            data-prefix="far"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              class=""
              d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200zm101.8-262.2L295.6 256l62.2 62.2c4.7 4.7 4.7 12.3 0 17l-22.6 22.6c-4.7 4.7-12.3 4.7-17 0L256 295.6l-62.2 62.2c-4.7 4.7-12.3 4.7-17 0l-22.6-22.6c-4.7-4.7-4.7-12.3 0-17l62.2-62.2-62.2-62.2c-4.7-4.7-4.7-12.3 0-17l22.6-22.6c4.7-4.7 12.3-4.7 17 0l62.2 62.2 62.2-62.2c4.7-4.7 12.3-4.7 17 0l22.6 22.6c4.7 4.7 4.7 12.3 0 17z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
    </div>
  </form>
</div>
`;

exports[`TaskBase.vue Matches snapshot 1`] = `
<div>
  <div
    class="row no-gutters"
  >
    <div
      class="col-1 checkbox-wrapper"
    >
      <svg
        aria-hidden="true"
        class="pointer mr-3 svg-inline--fa fa-square fa-w-14"
        data-icon="square"
        data-prefix="far"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          class=""
          d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm-6 400H54c-3.3 0-6-2.7-6-6V86c0-3.3 2.7-6 6-6h340c3.3 0 6 2.7 6 6v340c0 3.3-2.7 6-6 6z"
          fill="currentColor"
        />
      </svg>
       
      <!---->
       
      <!---->
    </div>
     
    <div
      class="col"
    >
      <div
        class="row no-gutters"
      >
        <div
          class="mr-3 col-md-1 col-2"
        >
          <span
            class="badge badge--task badge-info"
          >
            
          </span>
        </div>
         
        <div
          class="label-wrapper col"
        >
          
            
        
        </div>
      </div>
    </div>
     
    <div
      class="text-right col-2"
    >
      <svg
        aria-hidden="true"
        class="pointer mr-3 svg-inline--fa fa-pen fa-w-16 fa-sm"
        data-icon="pen"
        data-prefix="fas"
        data-test-id="open-editor"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          class=""
          d="M290.74 93.24l128.02 128.02-277.99 277.99-114.14 12.6C11.35 513.54-1.56 500.62.14 485.34l12.7-114.22 277.9-277.88zm207.2-19.06l-60.11-60.11c-18.75-18.75-49.16-18.75-67.91 0l-56.55 56.55 128.02 128.02 56.55-56.55c18.75-18.76 18.75-49.16 0-67.91z"
          fill="currentColor"
        />
      </svg>
       
      <svg
        aria-hidden="true"
        class="pointer svg-inline--fa fa-trash-alt fa-w-14"
        data-icon="trash-alt"
        data-prefix="far"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          class=""
          d="M268 416h24a12 12 0 0 0 12-12V188a12 12 0 0 0-12-12h-24a12 12 0 0 0-12 12v216a12 12 0 0 0 12 12zM432 80h-82.41l-34-56.7A48 48 0 0 0 274.41 0H173.59a48 48 0 0 0-41.16 23.3L98.41 80H16A16 16 0 0 0 0 96v16a16 16 0 0 0 16 16h16v336a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128h16a16 16 0 0 0 16-16V96a16 16 0 0 0-16-16zM171.84 50.91A6 6 0 0 1 177 48h94a6 6 0 0 1 5.15 2.91L293.61 80H154.39zM368 464H80V128h288zm-212-48h24a12 12 0 0 0 12-12V188a12 12 0 0 0-12-12h-24a12 12 0 0 0-12 12v216a12 12 0 0 0 12 12z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
   
  <div>
    <!---->
  </div>
</div>
`;

exports[`ToDoBase.vue Matches snapshot 1`] = `
<div>
  <h5>
    
    ToDo List
  
  </h5>
   
  <div
    class="mb-2"
  >
    
    Monday, December 9th 2019
  
  </div>
   
  <div>
    <!---->
     
    <div
      class="incomplete-task-list mb-2"
    >
      <div
        class="text-muted"
        style="display: none;"
      >
        <small>
          <i>
            All tasks complete
          </i>
        </small>
      </div>
       
      <div
        class="task-wrapper"
      >
        <div>
          <div
            class="row no-gutters"
          >
            <div
              class="col-1 checkbox-wrapper"
            >
              <svg
                aria-hidden="true"
                class="pointer mr-3 svg-inline--fa fa-square fa-w-14"
                data-icon="square"
                data-prefix="far"
                focusable="false"
                role="img"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  class=""
                  d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm-6 400H54c-3.3 0-6-2.7-6-6V86c0-3.3 2.7-6 6-6h340c3.3 0 6 2.7 6 6v340c0 3.3-2.7 6-6 6z"
                  fill="currentColor"
                />
              </svg>
               
              <!---->
               
              <!---->
            </div>
             
            <div
              class="col"
            >
              <div
                class="row no-gutters"
              >
                <div
                  class="mr-3 col-md-1 col-2"
                >
                  <span
                    class="badge badge--task badge-info"
                  >
                    Task
                  </span>
                </div>
                 
                <div
                  class="label-wrapper col"
                >
                  
            Walk the dog again
        
                </div>
              </div>
            </div>
             
            <div
              class="text-right col-2"
            >
              <svg
                aria-hidden="true"
                class="pointer mr-3 svg-inline--fa fa-pen fa-w-16 fa-sm"
                data-icon="pen"
                data-prefix="fas"
                data-test-id="open-editor"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  class=""
                  d="M290.74 93.24l128.02 128.02-277.99 277.99-114.14 12.6C11.35 513.54-1.56 500.62.14 485.34l12.7-114.22 277.9-277.88zm207.2-19.06l-60.11-60.11c-18.75-18.75-49.16-18.75-67.91 0l-56.55 56.55 128.02 128.02 56.55-56.55c18.75-18.76 18.75-49.16 0-67.91z"
                  fill="currentColor"
                />
              </svg>
               
              <svg
                aria-hidden="true"
                class="pointer svg-inline--fa fa-trash-alt fa-w-14"
                data-icon="trash-alt"
                data-prefix="far"
                focusable="false"
                role="img"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  class=""
                  d="M268 416h24a12 12 0 0 0 12-12V188a12 12 0 0 0-12-12h-24a12 12 0 0 0-12 12v216a12 12 0 0 0 12 12zM432 80h-82.41l-34-56.7A48 48 0 0 0 274.41 0H173.59a48 48 0 0 0-41.16 23.3L98.41 80H16A16 16 0 0 0 0 96v16a16 16 0 0 0 16 16h16v336a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128h16a16 16 0 0 0 16-16V96a16 16 0 0 0-16-16zM171.84 50.91A6 6 0 0 1 177 48h94a6 6 0 0 1 5.15 2.91L293.61 80H154.39zM368 464H80V128h288zm-212-48h24a12 12 0 0 0 12-12V188a12 12 0 0 0-12-12h-24a12 12 0 0 0-12 12v216a12 12 0 0 0 12 12z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </div>
           
          <div>
            <!---->
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="complete-task-list mb-4"
    >
      <div>
        <div
          class="task"
        >
          <div
            class="row no-gutters"
          >
            <div
              class="col-1 checkbox-wrapper"
            >
              <!---->
               
              <svg
                aria-hidden="true"
                class="pointer mr-3 svg-inline--fa fa-check-square fa-w-14"
                data-icon="check-square"
                data-prefix="far"
                focusable="false"
                role="img"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  class=""
                  d="M400 32H48C21.49 32 0 53.49 0 80v352c0 26.51 21.49 48 48 48h352c26.51 0 48-21.49 48-48V80c0-26.51-21.49-48-48-48zm0 400H48V80h352v352zm-35.864-241.724L191.547 361.48c-4.705 4.667-12.303 4.637-16.97-.068l-90.781-91.516c-4.667-4.705-4.637-12.303.069-16.971l22.719-22.536c4.705-4.667 12.303-4.637 16.97.069l59.792 60.277 141.352-140.216c4.705-4.667 12.303-4.637 16.97.068l22.536 22.718c4.667 4.706 4.637 12.304-.068 16.971z"
                  fill="currentColor"
                />
              </svg>
               
              <!---->
            </div>
             
            <div
              class="col"
            >
              <div
                class="row no-gutters"
              >
                <div
                  class="mr-3 col-md-1 col-2"
                >
                  <span
                    class="badge badge--task badge-info"
                  >
                    Task
                  </span>
                </div>
                 
                <div
                  class="label-wrapper col"
                >
                  
            Walk the dog
        
                </div>
              </div>
            </div>
             
            <div
              class="text-right col-2"
            >
              <svg
                aria-hidden="true"
                class="pointer mr-3 svg-inline--fa fa-pen fa-w-16 fa-sm"
                data-icon="pen"
                data-prefix="fas"
                data-test-id="open-editor"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  class=""
                  d="M290.74 93.24l128.02 128.02-277.99 277.99-114.14 12.6C11.35 513.54-1.56 500.62.14 485.34l12.7-114.22 277.9-277.88zm207.2-19.06l-60.11-60.11c-18.75-18.75-49.16-18.75-67.91 0l-56.55 56.55 128.02 128.02 56.55-56.55c18.75-18.76 18.75-49.16 0-67.91z"
                  fill="currentColor"
                />
              </svg>
               
              <svg
                aria-hidden="true"
                class="pointer svg-inline--fa fa-trash-alt fa-w-14"
                data-icon="trash-alt"
                data-prefix="far"
                focusable="false"
                role="img"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  class=""
                  d="M268 416h24a12 12 0 0 0 12-12V188a12 12 0 0 0-12-12h-24a12 12 0 0 0-12 12v216a12 12 0 0 0 12 12zM432 80h-82.41l-34-56.7A48 48 0 0 0 274.41 0H173.59a48 48 0 0 0-41.16 23.3L98.41 80H16A16 16 0 0 0 0 96v16a16 16 0 0 0 16 16h16v336a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128h16a16 16 0 0 0 16-16V96a16 16 0 0 0-16-16zM171.84 50.91A6 6 0 0 1 177 48h94a6 6 0 0 1 5.15 2.91L293.61 80H154.39zM368 464H80V128h288zm-212-48h24a12 12 0 0 0 12-12V188a12 12 0 0 0-12-12h-24a12 12 0 0 0-12 12v216a12 12 0 0 0 12 12z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </div>
           
          <div>
            <!---->
          </div>
        </div>
      </div>
    </div>
     
    <div>
      <div>
        <div
          class="text-md-left text-center"
        >
          <svg
            aria-hidden="true"
            class="pointer add-button svg-inline--fa fa-plus-square fa-w-14 fa-2x"
            data-icon="plus-square"
            data-prefix="far"
            focusable="false"
            role="img"
            style="display: none;"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              class=""
              d="M352 240v32c0 6.6-5.4 12-12 12h-88v88c0 6.6-5.4 12-12 12h-32c-6.6 0-12-5.4-12-12v-88h-88c-6.6 0-12-5.4-12-12v-32c0-6.6 5.4-12 12-12h88v-88c0-6.6 5.4-12 12-12h32c6.6 0 12 5.4 12 12v88h88c6.6 0 12 5.4 12 12zm96-160v352c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V80c0-26.5 21.5-48 48-48h352c26.5 0 48 21.5 48 48zm-48 346V86c0-3.3-2.7-6-6-6H54c-3.3 0-6 2.7-6 6v340c0 3.3 2.7 6 6 6h340c3.3 0 6-2.7 6-6z"
              fill="currentColor"
            />
          </svg>
        </div>
         
        <form
          class="mt-2"
        >
          <input
            aria-required="true"
            class="mb-2 form-control"
            id="__BVID__19"
            placeholder="What do you need to do?"
            required="required"
            test-data-name=""
            type="text"
          />
           
          <div
            class="mb-2 custom-control custom-radio"
          >
            <input
              aria-required="true"
              autocomplete="off"
              class="custom-control-input"
              id="__BVID__20"
              name="some-radios"
              required="required"
              test-data-radiotask=""
              type="radio"
              value="task"
            />
            <label
              class="custom-control-label"
              for="__BVID__20"
            >
              Task
            </label>
          </div>
           
          <div
            class="mb-2 custom-control custom-radio"
          >
            <input
              aria-required="true"
              autocomplete="off"
              class="custom-control-input"
              id="__BVID__21"
              name="some-radios"
              required="required"
              type="radio"
              value="event"
            />
            <label
              class="custom-control-label"
              for="__BVID__21"
            >
              Event
            </label>
          </div>
           
          <div
            class="row mt-3 no-gutters"
          >
            <div
              class="col"
            >
              <button
                class="btn btn-sm btn-primary"
                type="submit"
              >
                Add Task
              </button>
            </div>
             
            <div
              class="text-right col"
            >
              <svg
                aria-hidden="true"
                class="pointer svg-inline--fa fa-times-circle fa-w-16 fa-2x"
                data-icon="times-circle"
                data-prefix="far"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  class=""
                  d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200zm101.8-262.2L295.6 256l62.2 62.2c4.7 4.7 4.7 12.3 0 17l-22.6 22.6c-4.7 4.7-12.3 4.7-17 0L256 295.6l-62.2 62.2c-4.7 4.7-12.3 4.7-17 0l-22.6-22.6c-4.7-4.7-4.7-12.3 0-17l62.2-62.2-62.2-62.2c-4.7-4.7-4.7-12.3 0-17l22.6-22.6c4.7-4.7 12.3-4.7 17 0l62.2 62.2 62.2-62.2c4.7-4.7 12.3-4.7 17 0l22.6 22.6c4.7 4.7 4.7 12.3 0 17z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
